### PR TITLE
Debugger 9000

### PR DESCRIPTION
**Pull request by pelikhan**
Test it: https://maker.makecode.com/app/7cbac7abf891b2b2c6d68b6b9d04c296e48d2161-6082f7e3cd
- [x] slow-mo replaced by debugger toggle button (needs better look)
- [x] remove stop button (debugger toggle button)
- [x] move slow-mo in debugger toolbar
- [x] disable toolbar start/stop/restart while debugging
- [x] auto step-into until a block/text is found
- [x] reconcile debugging with tracing: we should be able trace and pause (@riknoll @mmoskal  I'll need your help on this)
- [x] typescript stepping
- [x] in blocks, ensure that all steps land on a block?
- [x] move category table under sliding menu
- [x] add yellow halo around modified variable
clip

### CODE REVIEW

**Comment by samelhusseini** [comment]
Yes that sounds better actually, you have the “play mode” (first button) which is doing tracing. once you pause, you can step, and then restart.

________________________________

**Comment by abchatra** [comment]
Can we have a staged release for this feature? Lets enable in one target first (arcade\maker makes most sense for me) and iterate. Documentation and user testing are required before we enable this broadly.

**Comment by pelikhan** [comment]
Usage of tracing is quite low but sure we can keep the flag

________________________________

**Comment by pelikhan** [comment]
@abchatra tracing is back if "enableTrace" is true.

File: docs/targets/pxtarget.md [hunk 1]
```diff
@@ -221,7 +221,6 @@ Each target has one board (plus optional parts).
         // buttons and parts
         hideRestart?: boolean;      // hide the restart button 
         hideFullscreen?: boolean;   // hide the fullscreen button
-        enableTrace?: boolean;      // enable the slow-mode (snail) button
```
Comment (samelhusseini) [inline docs/targets/pxtarget.md]: You still need this.

**Pull request merged**